File: libraryCommon/src/main/java/com/sunny/common/injection/scope/ConfigPersistent.java
package com.sunny.common.injection.scope;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Scope;

/**
 * A scoping annotation to permit dependencies conform to the life of the
 * {ConfigPersistentComponent}
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface ConfigPersistent {
}

File: app/src/main/java/uk/co/ribot/androidboilerplate/injection/scope/ConfigPersistent.kt
package uk.co.ribot.androidboilerplate.injection.scope

import uk.co.ribot.androidboilerplate.injection.component.ConfigPersistentComponent
import javax.inject.Scope

/**
 * A scoping annotation to permit dependencies conform to the life of the
 * [ConfigPersistentComponent]
 */
@Scope
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class ConfigPersistent
